simulate-switching-workspaces-on-active-monitor@micheledaros.com/extension.js

```diff
@@ -142,7 +133,9 @@ class WorkSpacesService {
     }
 
     _getFocusedMonitor() {
-        return global.display.get_current_monitor();
+        let currentMonitor = global.display.get_current_monitor();
+        maybeLog(` focus is currently on monitor ${currentMonitor} `);
```

Comment:
HI @Wizzerinus is this the missing API? On which version of gnome?
On my machine (Ubuntu with Gnome 45.1) it works.

![log](https://github.com/micheledaros/gnome-shell-extension-simulate-switching-workspaces-on-active-monitor/assets/19147654/c2ddce07-9ee7-4156-84de-3d087bf4dff6)

Reply:
Or by "missing API" do you mean that `global.display.get_current_monitor()` used to return the index of monitor with the window on focus, and with gnome 45  returns the monitor with the mouse pointer?

Reply:
The missing API is being unable to override `Main.activateWindow`. As far as I understand it was used as follows:
- Have a window 1 on monitor 1 and a window 2 on monitor 2
- Focus window 1, mouse pointer within it
- Alt tab into window 2, mouse pointer within windows 1
Now `get_current_monitor` returns monitor 1, despite the focus being within monitor 2. Before the update, activateWindow would override it.

Reply:
It felt annoying but I'm not sure how bad it actually is

Reply:
Got it, thanks. 
Let's merge it the way it is, since it's better than having no extension at all, and try to address the issue in another iteration